pkg/scheduler/factory.go

```diff
@@ -113,181 +109,104 @@ type Config struct {
 
-	c := &Configurator{
-		client:                         args.Client,
-		informerFactory:                args.InformerFactory,
-		pVLister:                       args.PvInformer.Lister(),
-		pVCLister:                      args.PvcInformer.Lister(),
-		serviceLister:                  args.ServiceInformer.Lister(),
-		controllerLister:               args.ReplicationControllerInformer.Lister(),
-		replicaSetLister:               args.ReplicaSetInformer.Lister(),
-		statefulSetLister:              args.StatefulSetInformer.Lister(),
-		pdbLister:                      pdbLister,
-		nodeLister:                     args.NodeInformer.Lister(),
-		podLister:                      args.PodInformer.Lister(),
-		storageClassLister:             storageClassLister,
-		csiNodeLister:                  csiNodeLister,
-		volumeBinder:                   args.VolumeBinder,
-		schedulerCache:                 args.SchedulerCache,
-		StopEverything:                 stopEverything,
-		hardPodAffinitySymmetricWeight: args.HardPodAffinitySymmetricWeight,
-		disablePreemption:              args.DisablePreemption,
-		percentageOfNodesToScore:       args.PercentageOfNodesToScore,
-		bindTimeoutSeconds:             args.BindTimeoutSeconds,
-		podInitialBackoffSeconds:       args.PodInitialBackoffSeconds,
-		podMaxBackoffSeconds:           args.PodMaxBackoffSeconds,
-		enableNonPreempting:            utilfeature.DefaultFeatureGate.Enabled(features.NonPreemptingPriority),
-		registry:                       args.Registry,
-		plugins:                        args.Plugins,
-		pluginConfig:                   args.PluginConfig,
-		pluginConfigProducerRegistry:   args.PluginConfigProducerRegistry,
-		nodeInfoSnapshot:               nodeinfosnapshot.NewEmptySnapshot(),
-	}
 	c.factoryArgs = &PluginFactoryArgs{
 		NodeInfoLister:                 c.nodeInfoSnapshot.NodeInfos(),
 		PodLister:                      c.nodeInfoSnapshot.Pods(),
-		ServiceLister:                  c.serviceLister,
-		ControllerLister:               c.controllerLister,
-		ReplicaSetLister:               c.replicaSetLister,
-		StatefulSetLister:              c.statefulSetLister,
-		PDBLister:                      c.pdbLister,
-		CSINodeLister:                  c.csiNodeLister,
-		PVLister:                       c.pVLister,
-		PVCLister:                      c.pVCLister,
-		StorageClassLister:             c.storageClassLister,
-		VolumeBinder:                   c.volumeBinder,
-		HardPodAffinitySymmetricWeight: c.hardPodAffinitySymmetricWeight,
+		ServiceLister:                  c.ServiceLister,
+		ControllerLister:               c.ControllerLister,
+		ReplicaSetLister:               c.ReplicaSetLister,
+		StatefulSetLister:              c.StatefulSetLister,
+		PDBLister:                      c.PdbLister,
+		CSINodeLister:                  c.CSINodeLister,
+		PVLister:                       c.PVLister,
+		PVCLister:                      c.PVCLister,
+		StorageClassLister:             c.StorageClassLister,
```

Comment:
we can replace all of those with two arguments: scheduler's ```SharedLister``` and ```SharedInformerFactory```, perhaps we can do that in a followup PR.

Reply:
+1 on followup. That change would be less self-contained